### ISSUE TITLE
design and document REST API

### ISSUE THREAD

**Issue by muatik** [SWE574-RenameMe/swe574-CoolName#26]
Front-end, back-end, and mobile applications will work and be developed independently. For FE and mobile clients, the back-end app will expose a REST API.

This API should be designed according to the best practices. That is, it should be stateless and resourceful. 

How to design an API, how to work with the API-FIRST principle 
1. https://www.ckl.io/blog/api-design-think-first-code-later/
2. https://blog.qmo.io/ultimate-guide-to-api-design/
3. https://hackernoon.com/restful-api-designing-guidelines-the-best-practices-60e1d954e7c9#.fduz2pa6i
4. https://www.infoq.com/articles/paypal-api-first-part2

We will go through all the mockup pages and design APIs that are required to implement each of the mockup pages.

### Documentation
API documentation should be available on Wiki pages. Each API request needs to be clearly defined and needs to have at least one request and response example.

- [x] Design
- [ ] Document

**Comment by muatik** [comment]
We have almost finished the API design and documented it onto a single wiki page. https://github.com/SWE574-Groupago/heritago/wiki/REST-API The API documentation needs to be split into separate pages. And we may use one of the modern JSON documentation and mock server tools, such as API blueprint. https://apiblueprint.org/

In addition, to test rest API, we can prepare a postman project template and share it via the git repository. 

In short, I suggest using https://apiblueprint.org/ It facilitates API related works and definitely reduce the amount of communication between FE and BE teams.

**Comment by fkucuk** [comment]
Nice resources @muatik .
And I just wanted to remind, the API should follow the  [W3's web annotation protocols](https://www.w3.org/TR/annotation-protocol/)

**Comment by fkucuk** [comment]
about "POST /heritages/1/annotations", I made some contribution to wiki. I've added some detail to support the web annotation protocol.
And since the api should act like a separated annotation vault,  the endpoint name should be different. I think the path /annotations is enough. Or we can both support "/heritages/1/annotations" and "/annotations" by redirecting one to another.

**Comment by muatik** [comment]
@fkucuk `/annotations` could be another way to access to annotations. However, how do plan to fetch annotations made for a specific resource, can you give an example?
